package.json

```diff
@@ -176,15 +177,20 @@
     "react-markdown": "2.4.2",
     "react-redux": "4.4.5",
     "react-router": "2.0.0",
+    "react-router-dom": "4.1.2",
     "react-router-redux": "4.0.4",
     "react-select": "1.0.0-rc.1",
     "react-sortable": "1.1.0",
     "react-test-renderer": "15.6.1",
     "react-toggle": "3.0.1",
     "reactcss": "1.0.7",
+    "recompose": "0.25.0",
     "redux": "3.0.0",
+    "redux-actions": "2.2.1",
     "redux-thunk": "0.1.0",
+    "redux-thunks": "0.3.0",
```

Comment:
I started down the path of using `redux-thunks` but got rid of it because I think it added complexity without any benefit.  Also, it was causing a really hard to debug issue with the optimizer running into a syntax issue when building kibana via `npm run build`, but not when running `npm start`.  Never was able to track down exactly what the problem was but getting rid of this package + the style it insists on solved the problem.  I haven't had to use the thunk functions in a reducer the way the return value of handleActions is used so I don't see the benefit of it.

Reply:
v0.3.1 should hopefully fix the build issue. Bumping here would probably be a good idea.

@Stacey-Gammon I'm curious if you're doing async operations in Redux, and if so, how that looks.

Reply:
@w33ble I have some explorations here: https://github.com/elastic/x-pack-kibana/compare/master...Stacey-Gammon:feature/workspaces-ui#diff-8514854ef5e2eacba3f1ee1a68d4b1f6

Also here: https://github.com/elastic/kibana/pull/14518/files#diff-73fcabf5f53de2e56a50e909044214baR32

Reply:
That first link doesn't work, the second one does though. So raw thunks and splitting actions into success/pending/error. There's some abstractions around that that I prefer personally, since it reduces the redux boilerplate a lot, but I can also appreciate trying to be explicit.